velox/common/memory/MemoryPool.h

```diff
@@ -544,15 +544,22 @@ std::ostream& operator<<(std::ostream& os, const MemoryPool::Stats& stats);
 
 class MemoryPoolImpl : public MemoryPool {
  public:
+  /// The callback invoked on the root memory pool destruction. It is set by
+  /// memory manager to return back the allocated memory capacity.
   using DestructionCallback = std::function<void(MemoryPool*)>;
+  /// The callback invoked when the used memory reservation of the root memory
+  /// pool exceed its capacity. It is set by memory manager to grow the memory
+  /// pool capacity.
+  using GrowCapacityCallback = std::function<bool(MemoryPool*, uint64_t)>;
```

Comment:
/// It should return false when the operation in callback fails. When it returns false, the capacity grow will fail, too.